### PR TITLE
Upgrade to PHPUnit 10 🥳️

### PR DESCRIPTION
**Pull request by TomasVotruba**
- Upgrade to PHPUnit 10
- bump to PHPUnit 10

### CODE REVIEW

**Comment by rvanlaak** [comment]
So we need to contribute with some rectors to modify to static and to classes renaming. Sounds feasible 😉

**Comment by TomasVotruba** [comment]
This is a great summary of the upgrade path :)
https://getrector.com/blog/how-to-upgrade-to-phpunit-10-in-diffs

File: composer.json [hunk 1]
```diff
@@ -39,7 +39,6 @@
         "webmozart/assert": "^1.11"
     },
     "require-dev": {
-        "brianium/paratest": "^6.7",
```
Comment (OskarStark) [inline composer.json:42]: 7.0 was released yesterday and support PHPUnit 10 now 👍

Comment (samsonasik) [reply composer.json:42]: it already rolled back :) https://github.com/rectorphp/rector-src/blob/9ec3967a5bb51519dec381255b7c6c0659a41d6e/composer.json#L43, see PR:

- https://github.com/rectorphp/rector-src/pull/3335

**Pull request merged**